Formula/transmission-cli.rb

```diff
@@ -24,31 +20,42 @@ class TransmissionCli < Formula
     sha256 x86_64_linux:   "178d05964e9efd8d4541cf5589f4772ccdc59b8de83158f96b8ad7ffeff6b8d2"
   end
 
+  depends_on "cmake" => :build
+  depends_on "gettext" => :build
+  depends_on "ninja" => :build
   depends_on "pkg-config" => :build
   depends_on "libevent"
   depends_on "openssl@1.1"
 
   uses_from_macos "curl"
   uses_from_macos "zlib"
 
+  on_linux do
+    depends_on "python@3.11" => :build
+  end
+
   def install
+    args = %W[
+      -DCMAKE_BUILD_TYPE=RelWithDebInfo
+      -DCMAKE_INSTALL_PREFIX=#{prefix}
+      -DENABLE_DAEMON=ON
+      -DENABLE_QT=OFF
+      -DENABLE_WEB=OFF
+      -DENABLE_UTILS=ON
+      -DENABLE_CLI=ON
+      -DENABLE_MAC=OFF
+      -DENABLE_TESTS=OFF
+      -DENABLE_NLS=OFF
+    ]
+
     if OS.mac?
       ENV.append "LDFLAGS", "-framework Foundation -prebind"
       ENV.append "LDFLAGS", "-liconv"
+      args << "-DCMAKE_OSX_SYSROOT=#{MacOS.sdk_path}"
     end
 
-    args = %W[
-      --disable-dependency-tracking
-      --prefix=#{prefix}
-      --disable-mac
-      --disable-nls
-      --without-gtk
-    ]
-
-    system "./configure", *args
-    system "make", "install"
-
-    (var/"transmission").mkpath
+    system "cmake", "-S", ".", "-B", "obj", "-G", "Ninja", *args
```

Comment:
Why ninja?

Reply:
I'm not entirely sure. I struggled to get this to build without this, but i'm definitely no expert so perhaps it can be done without.

I'm using this CI action from transmission as a basis for the formula: https://github.com/transmission/transmission/blob/671f835228c3092d75df4923ef8e4cd4835eb053/.github/workflows/actions.yml#L204